src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeInfo.cs

```diff
@@ -159,10 +163,13 @@ public static Task<SyntaxTreeDeclarationInfo> GetDeclarationInfoAsync(Document d
                             containsElementAccess = containsElementAccess || syntaxFacts.IsElementAccessExpression(node);
                             containsIndexerMemberCref = containsIndexerMemberCref || syntaxFacts.IsIndexerMemberCRef(node);
 
-                            DeclaredSymbolInfo declaredSymbolInfo;
-                            if (syntaxFacts.TryGetDeclaredSymbolInfo(node, out declaredSymbolInfo))
+                            if (!isGeneratedCode)
                             {
-                                declaredSymbolInfos.Add(declaredSymbolInfo);
+                                DeclaredSymbolInfo declaredSymbolInfo;
+                                if (syntaxFacts.TryGetDeclaredSymbolInfo(node, out declaredSymbolInfo))
+                                {
+                                    declaredSymbolInfos.Add(declaredSymbolInfo);
```

Comment:
This code path is hit by other features that need to see all syntax trees/nodes and only NavigateTo along with its support structure `DeclaredSymbolInfo` can safely skip this processing.